### PR TITLE
ocamlPackages.faraday: 0.8.1 → 0.8.2

### PR DESCRIPTION
**Pull request by vbgl**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/ocaml-modules/faraday/default.nix [hunk 1]
```diff
@@ -2,22 +2,18 @@
 
 buildDunePackage rec {
   pname = "faraday";
-  version = "0.8.1";
-
-  useDune2 = true;
-
-  minimumOCamlVersion = "4.02";
```
Comment (Et7f3) [inline pkgs/development/ocaml-modules/faraday/default.nix:9]: Why are you removing this ? does it build with 4.0 ?

**Review by Et7f3 — APPROVED** [review]
` nix build github:vbgl/nixpkgs/ocaml-faraday-0.8.2#ocaml-ng.ocamlPackages_4_08.faraday` and ` nix build github:vbgl/nixpkgs/ocaml-faraday-0.8.2#ocaml-ng.ocamlPackages_4_08.faraday` compile fine on x86_64-darwin. However I can't post result of Nixpkgs-reviews it seem to be unrelated breakage

File: pkgs/development/ocaml-modules/faraday/default.nix (hunk 1 above)
Comment (vbgl) [reply pkgs/development/ocaml-modules/faraday/default.nix:9]: What do you mean by “better”?

Comment (Et7f3) [reply pkgs/development/ocaml-modules/faraday/default.nix:9]:
```
$  nix build github:vbgl/nixpkgs/ocaml-faraday-0.8.2#ocaml-ng.ocamlPackages_4_07.faraday
error: fmt is not available for OCaml 4.07.1
```
if we set this to `4.08.0` then we get `error: faraday-0.8.2 is not available for OCaml 4.07.1`

Comment (vbgl) [reply pkgs/development/ocaml-modules/faraday/default.nix:9]: That’s right. But you don’t explain why you think one of those is better than the other.

Comment (Et7f3) [reply pkgs/development/ocaml-modules/faraday/default.nix:9]: In the first case as users that want package `faraday` I receive error message that cite `fmt` from user point of view it is unrelated unless it dig and see it is a dependencies. I find more explicit but you can ignore if you are not convinced.

Comment (vbgl) [reply pkgs/development/ocaml-modules/faraday/default.nix:9]: Fixed. Thanks.

**Review by Et7f3 — APPROVED** [review]

**Comment by Et7f3** [comment]
If I apply the two linked PR this PR build fine.

**Comment by vbgl** [comment]
@GrahamcOfBorg build ocamlPackages.faraday

**Pull request merged**